src/components/CodeExample.js

```diff
@@ -130,28 +125,55 @@ const CodeSnippet = ({ codeSnippets, title, href }) => {
   );
   const [activeLang, setActiveLang] = React.useState("");
   const [isCopied, setCopy] = React.useState(false);
+  const [isHovered, setHover] = React.useState(false);
+  const [dimension, setDimension] = React.useState({ right: 0, top: 0 });
+
   const codeSnippetsArr = React.Children.toArray(codeSnippets);
   const selectedSnippet =
     codeSnippetsArr.find(
       (snippet) => snippet.props["data-language"] === activeLang,
     ) || codeSnippetsArr[0];
   const SelectedSnippetStr = mdxJsxToString(selectedSnippet);
 
+  const CopyToClipboardRef = React.useCallback((node) => {
+    if (node !== null) {
+      setDimension({
+        right: Math.floor(node.getBoundingClientRect().right),
+        top: Math.floor(node.getBoundingClientRect().top),
+      });
+    }
+  }, []);
+
   const onChange = React.useCallback((e) => {
     const { value } = e.target;
     document.cookie = `lang=${value}`;
-    navigate(`?lang=${value}`, { replace: true });
+    navigate(`?lang=${value}`, {
+      replace: true,
+    });
     setActiveLang(value);
   }, []);
 
   React.useEffect(() => {
     const langCookie = getCookie("lang");
+    let timer;
     if (langCookie) {
       setActiveLang(langCookie);
     } else {
       setActiveLang(availableLangs[0]);
     }
-  }, [availableLangs]);
+
+    if (isCopied) {
```

Comment:
Once `CopyToClipboard` icon gets clicked, `isCopied` is set to true then its value gets set to false in order to show its previous text `Click to Copy`